### ISSUE TITLE
Pydantic 2.0 (complete rewrite) breaks copier

### ISSUE THREAD

**Issue by henryiii** [copier-org/copier#1225]
### Describe the problem

After over a year and two PyCons, the Rust rewrite of Pydandic with 17x+ speedup (last I heard maybe it was 23x?) is out (four hours ago)! Unfortunately, it breaks copier. As much as I [hate suggesting it](https://iscinumpy.dev/post/bound-version-constraints/), you might want to do a quick patch release with a version cap, then work on making it work (Unless it's really easy to fix, of course, which it might be).

Also, it's a really bad idea to use things from dependencies that start with underscores. Just saying. :P

(They did an incredible amount of work to try to make sure they didn't break users. Users importing underscore objects doesn't count!)

### Template

https://github.com/scientific-python/cookie

(If you want one, that is - you can't even start copier on a fresh install)

### To Reproduce

Run copier.

That's all.

### Logs

```pytb
  File "/home/runner/work/cookie/cookie/.nox/compare_copier/bin/copier", line 5, in <module>
    from copier.__main__ import copier_app_run
  File "/home/runner/work/cookie/cookie/.nox/compare_copier/lib/python3.8/site-packages/copier/__init__.py", line 6, in <module>
    from .main import *  # noqa: F401,F403
  File "/home/runner/work/cookie/cookie/.nox/compare_copier/lib/python3.8/site-packages/copier/main.py", line 29, in <module>
    from .errors import (
  File "/home/runner/work/cookie/cookie/.nox/compare_copier/lib/python3.8/site-packages/copier/errors.py", line 6, in <module>
    from pydantic.errors import _PathValueError
ImportError: cannot import name '_PathValueError' from 'pydantic.errors' (/home/runner/work/cookie/cookie/.nox/compare_copier/lib/python3.8/site-packages/pydantic/errors.py)
```

### Expected behavior

Not to break when pydantic is installed, or not to install pedantic 2.0.

### Operating system

macOS

### Operating system distribution and version

Any

### Copier version

8.0.0

### Python version

All

### Installation method

pipx+pypi

### Additional context

Broken logs at https://github.com/scientific-python/cookie/actions/runs/5426746583/jobs/9869183393

**Comment by luismata01** [comment]
Hi
I have the same problem use this ti install copier on Doodba copier template

```

Traceback (most recent call last):
File "/home/luis/.local/bin/copier", line 5, in
from copier.main import copier_app_run
File "/home/luis/.local/pipx/venvs/copier/lib/python3.8/site-packages/copier/init.py", line 6, in
from .main import * # noqa: F401,F403
File "/home/luis/.local/pipx/venvs/copier/lib/python3.8/site-packages/copier/main.py", line 29, in
from .errors import (
File "/home/luis/.local/pipx/venvs/copier/lib/python3.8/site-packages/copier/errors.py", line 6, in
from pydantic.errors import _PathValueError
ImportError: cannot import name '_PathValueError' from 'pydantic.errors' (/home/luis/.local/pipx/venvs/copier/lib/python3.8/site-packages/pydantic/errors.py

```

https://github.com/Tecnativa/doodba-copier-template/issues/392#issue-1783821586

Can you help me  please

**Comment by luismata01** [comment]
> Member
Thank's for your support

This solution work form me using 
`pipx inject copier 'pydantic<2'`

Thanks

**Comment by roestzwiee** [comment]
Hi,
for those who have a dependency to copier v7, will there also be a fix for copier 7.2?

**Comment by henryiii** [comment]
> for those who have a dependency to copier v7, will there also be a fix for copier 7.2?

https://github.com/copier-org/copier/pull/1229#issuecomment-1617725867 seems to indicate that might happen.

**Comment by yajo** [comment]
You can enter a shell 

> As much as I [hate suggesting it](https://iscinumpy.dev/post/bound-version-constraints/), you might want to do a quick patch release with a version cap, then work on making it work

See https://github.com/copier-org/copier/pull/1244.
